### PR TITLE
Cleanup HttpPipelineMessage and HttpClientOptions

### PR DESCRIPTION
**Pull request by pakrym**
1. Remove IServiceProvider from client options and transport as we don't use it
3. Remove HttpMessageOptions as they are not used
2. Rename HttpPipelineMessage.Cancellation to CancellationToken for consistency

### Breaking change

Before:

``` C#
    class Policy : HttpPipelinePolicy
    {
        private async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
        {
            await DoSomething(message.Cancellation);
        }
    }
```

After:

``` C#
    class Policy : HttpPipelinePolicy
    {
        private async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
        {
            await DoSomething(message.CancellationToken);
        }
    }
```

### CODE REVIEW

File: sdk/storage/Azure.Storage.Queues/tests/MessageIdClientTests.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ public async Task UpdateAsync_Update()
                 var result = await messageId.UpdateAsync(message1, enqueuedMessage.PopReceipt, new TimeSpan(100));
 
                 // Assert
-                Assert.IsNotNull(result.Raw.Headers.RequestId);
+                Assert.IsNotNull(result.GetRawResponse().Headers.RequestId);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Queues/tests/MessageIdClientTests.cs]: Was this accidentally left in from the last change?

File: sdk/core/Azure.Core/src/Pipeline/HttpMessageOptions.cs [hunk 2]
```diff
@@ -1,62 +0,0 @@
...
-            internal HttpMessageOptions(HttpPipelineMessage message)
-                => _message = message;
-
-            public T Add<T>(T option) where T : class
-            {
-                _message._options.SetOption(typeof(T), option);
-                return option;
-            }
-
-            public void Add(object option)
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Pipeline/HttpMessageOptions.cs]: We don't use these today, but have we decided there aren't scenarios for customers?

I'm fine removing them now and adding them back later, if necessary, but I thought they'd be useful for crazy extensibility scenarios where you wanted to enable different pipeline policies to communicate.

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs [hunk 3]
```diff
@@ -1,42 +1,23 @@
...
-using System;
-using System.ComponentModel;
 using System.Threading;
 
 namespace Azure.Core.Pipeline
 {
-    public partial class HttpPipelineMessage  : IDisposable
+    public struct HttpPipelineMessage
     {
-        internal OptionsStore _options = new OptionsStore();
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs]: Is anybody using the OptionsStore type now?

**Review by tg-msft — APPROVED** [review]
I left a couple of questions, but I think the cleaning looks good.

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs (hunk 3 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs]: Nope.

File: sdk/storage/Azure.Storage.Queues/tests/MessageIdClientTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/storage/Azure.Storage.Queues/tests/MessageIdClientTests.cs]: yeah, undone.

File: sdk/core/Azure.Core/src/Pipeline/HttpMessageOptions.cs (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/HttpMessageOptions.cs]: It's an additive feature that's not being used, is not implemented efficiently and not tested. I'll bring it back in better shape when we find a use case.

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs (hunk 3 above)
Comment (tg-msft) [reply sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs]: Should we remove it too?

Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs]: Just did

**Review by jsquire — APPROVED** [review]

**Pull request merged**